polling_stations/apps/data_importers/base_importers.py

```diff
@@ -241,22 +242,79 @@ def get_station_hash(self, station):
 
     def check_station_point(self, station_record):
         if station_record["location"]:
-            try:
-                council = Council.objects.get(
-                    geography__geography__covers=station_record["location"]
+            self.check_in_council_bounds(station_record)
+            self.check_duplicate_location(station_record)
+
+    def check_duplicate_location(self, station_record):
+        stations_with_location_and_different_postcode = [
+            s
+            for s in self.stations.elements
+            if s.location and (s.postcode != station_record["postcode"])
+        ]
+
+        if not stations_with_location_and_different_postcode:
+            return
+
+        srids = [
+            GEOSGeometry(s.location).srid
+            for s in stations_with_location_and_different_postcode
+        ]
+        srid_to_use = max(srids, key=srids.count)
+
+        if srid_to_use == 4326:
+            threshold = 0.001
+        if srid_to_use == 27700:
+            threshold = 10
+
+        station_index = rtree.index.Index()
+        for i, station in enumerate(stations_with_location_and_different_postcode):
+            geom = GEOSGeometry(station.location).transform(srid_to_use, clone=True)
+            station_index.insert(i, (geom.x, geom.y, geom.x, geom.y))
+
+        record_geom = station_record["location"].transform(srid_to_use, clone=True)
+        nearest_ids = list(
+            station_index.intersection(
+                (
+                    record_geom.x - threshold,
+                    record_geom.y - threshold,
+                    record_geom.x + threshold,
+                    record_geom.y + threshold,
+                ),
+                1,
+            )
+        )
+        if nearest_ids:
```

Comment:
I prefer to flip the conditional and return early in this circumstance, to avoid nesting:
```
if not nearest_ids:
    return

for I in nearest_ids:
...
```

Reply:
:D I was thinking of your trick with this when I wrote line 255, but had forgotten by the time I got here.